text_extensions_for_pandas/resources/span_array.js

```diff
@@ -255,11 +300,151 @@ if(!window.SpanArray || window.SpanArray.VERSION < VERSION) {
+                const ids = event.target.getAttribute("data-ids").split(",").slice(0, -1)
+                Array.from(ids)
+                    .map(id_tag => {
+                        return id_tag.substring(1)
+                    })
+                    .forEach(id => {
+                        const entry = doc_table_body.querySelector(`tr[data-id="${id}"]`)
+                        entry.classList.remove("hover")
+                    })
+            }
+        }, true)
+
+        // Click disable/enable events
+
+        doc_table_body.addEventListener("click", (event) => {
+            const closest_control_button = event.target.closest("button[data-control]")
+            if(closest_control_button == undefined) return
+
+            const closest_tr = event.target.closest("tr")
+            if(closest_tr == undefined) return
+
+            const matching_span = doc_object.lookup_table[closest_tr.getAttribute("data-id")]
+
+            switch(closest_control_button.getAttribute("data-control")) {
+                case "visibility":
+                    {
+                        if(matching_span != undefined) matching_span.visible = !matching_span.visible
+                        source_spanarray.render()
+                    }
+                    break;
+                case "highlight":
+                    {
+                        if(matching_span != undefined) matching_span.highlighted = !matching_span.highlighted
+                        source_spanarray.render()
+                    }
+                    break;
+            }
+
+
+
+        }, true)
+
+        doc_text.addEventListener("click", (event) => {
+            const closest_mark = event.target.closest("mark")
+            if(closest_mark == undefined) return
+
+            // Preprocess ID string into a list of IDs
+            const ids = closest_mark.getAttribute("data-ids")
+                .split(",")
+                .slice(0, -1)
+                .map(id => {
+                    return id.substring(1)
+                })
+            
+            // If any of the connected IDs are highlighted, we set all spans in the list to not highlighted.
+            // Inversely, we want all spans highlighted if none were previously.
+
+            const highlighted_entry = ids.find(id => {
+                return doc_object.lookup_table[id].highlighted
+            })
+
+            const is_highlighted = (highlighted_entry != undefined)
+
+            ids.forEach(id => {
+                const span = doc_object.lookup_table[id]
+                if(span != undefined) span.highlighted = !is_highlighted
+            })
+
+            source_spanarray.render()
+        })
+    }
+} else {
+    // SpanArray JS is already defined and not an outdated copy
+    // Replace global SpanArray CSS with latest copy IFF global stylesheet is undefined
+
+    const global_stylesheet = document.head.querySelector("style.span-array-css")
```

Comment:
This code looks like a duplicate with the block of code at the top of this file. Maybe merge into a single function?

Reply:
Moved the matching constant declarations to outside the block to save some bytes and readability. Both of these global checks will be removed in the PR mentioned in the first comment response once scoping is switched to local and the style block is removed.